### PR TITLE
Store klarna customer token

### PR DESCRIPTION
**Pull request by DanielePalombo**
This PR is pretty inspired by [PR](https://github.com/solidusio-contrib/solidus_klarna_payments/pull/28), the main difference is that it stores the customer token on the payment source, being more in line with solidus.

### CODE REVIEW

**Review by MassimilianoLattanzio — APPROVED** [review]
LGTM

I did QA only without `tokenization` because Klarna needs to enable it to your account, and in my test account, the tokenization is not enabled.

**Comment by DanielePalombo** [comment]
> LGTM
> 
> I did QA only without `tokenization` because Klarna needs to enable it to your account, and in my test account, the tokenization is not enabled.

Thank you @MassimilianoLattanzio , I have QA'd it with tokenization and the right account, everything worked as expected.
- The payment was saved to the wallet
- The Payment is shown at the checkout payment step
- Selecting an existing payment the order can be completed without asking payment info again.

**Review by kennyadsl — APPROVED** [review]

File: app/models/spree/payment_method/klarna_credit.rb [hunk 1]
```diff
@@ -93,12 +91,6 @@ def try_void(payment)
       def spree_order(options)
         Spree::Order.find_by(number: options[:order_id].split("-").first)
       end
-
-      def tokenization_flow
```
Comment (vassalloandrea) [inline app/models/spree/payment_method/klarna_credit.rb:97]: You are right, now the customer token is saved on the payment source but a customer token shouldn't be created only if the user is logged in? If you enable the tokenization flow from the admin, the extension will use the intent TOKENIZE but the user is not logged in.

Comment (DanielePalombo) [reply app/models/spree/payment_method/klarna_credit.rb:97]: What is the problem with it? even when the user is not logged in, Klarna will generate the authorization token and the extension will convert it to the customer token, but the user won't be able to re-use it for other payments. Does it make sense for you?

Comment (DanielePalombo) [reply app/models/spree/payment_method/klarna_credit.rb:97]: @vassalloandrea considering your thought I have enabled limited tokenization only for not guest orders.

**Review by vassalloandrea — APPROVED** [review]
LGTM 👍

**Pull request merged**